packages/Uploader/src/components/FileList/FileList.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import { UploaderContext } from "../../Uploader";
+import { getNumberWithUnits } from "../../helpers";
+import File from "../File";
+
+const propTypes = {
+  maxFileSize: PropTypes.number.isRequired,
+  supportedMimeTypes: PropTypes.arrayOf(PropTypes.string).isRequired,
+};
+
+const defaultProps = {};
+
+export default function FileList({ supportedMimeTypes, maxFileSize }) {
+  const { files } = React.useContext(UploaderContext);
+  const I18n = useI18n();
+
+  function getFileErrorText(file) {
+    if (!file.isSizeValid) {
+      return I18n.t("uploader.errors.size", { maxFileSize: getNumberWithUnits(maxFileSize) });
+    }
+
+    if (!file.isTypeValid) {
+      return I18n.t("uploader.errors.type", { supportedTypes: supportedMimeTypes.join(", ") });
+    }
+
+    if (!file.isServerValid) {
+      return file.error.message;
+    }
+
+    return null;
+  }
+
+  return (
+    files &&
+    files.map(file => (
+      <File
```

Comment:
For your follow-up `a11y` PR, it would be nice to use `<ul>` + `<li>` to make this a semantic list.  Or, alternatively, `role=list` + `role=listitem`.